src/irr_v3d.h

```diff
@@ -28,3 +28,12 @@ typedef core::vector3d<double> v3d;
 typedef core::vector3d<s16> v3s16;
 typedef core::vector3d<u16> v3u16;
 typedef core::vector3d<s32> v3s32;
+
+// Calculate Manhattan distance between two vectors.
+// This is used in depth sorting of blocks, nodes and faces
+// as a faster alternative to Euclidian distance.
+template<class T>
+int manhattanDistance(const core::vector3d<T>& a, const core::vector3d<T>& b)
```

Comment:
nitpicking yes yes, shouldn't this return `u32`?

Reply:
Yes, manhattan distance is the sum of the absolute differences between the two vectors. It can't be negative.

Reply:
Is that really so much cheaper than just calculating the square of the euclidian distance (i.e. the dot product)?
(three multiplication instead of three subtractions and three abs')
Not a big deal, just curious.

Reply:
@sfan5 This should return T since T could also be a float.
@lhofhansl Made a [test](https://gist.github.com/x2048/93ebf986b2caea5bf45a36cf73a640d5) on some older hardware, integers are slightly better with squared euclidean, floats yield about 30% performance increase with manhattan.

Reply:
`T` is not a good idea, if you put in a `v3s16` with too high values there will be a signed overflow (UB) but more importantly the results will be wrong.

This can't happen for mapblock positions but still...
[This](https://en.cppreference.com/w/cpp/types/is_floating_point) could be used to provide a `T` return type for floats but `u32` for all others.

Reply:
Added results for AMD Ryzen and Raspberry PI 4 (ARM v7). Manhattan distance indeed does not provide any noticeable gain or is slower, except for old Intel Core i5 and float.

@sfan5 I'll remove manhattanDistance altogether to make the code simpler.